### PR TITLE
Add new article about constructor function and function context

### PR DESCRIPTION
**Pull request by caroleolivier**

### CODE REVIEW

File: _posts/2018-01-20-constructor-function-and-function-context-in-javascript.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+---
+layout: post
+date: 2018-01-20
+title: Constructor function and function context in JavaScript
```
Comment (simonproffitt) [inline _posts/2018-01-20-constructor-function-and-function-context-in-javascript.md]: I think this should probably be 'constructor function_s_ '

File: _posts/2018-01-20-constructor-function-and-function-context-in-javascript.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+---
+
+Today, I spent some time looking into JavaScript constructor functions. I ended up reading (again...) about function context and it finally sunk in. Here are my findings.
+
+
+
+#### Constructor Function
+
+First of all, what is a constructor function? A constructor function is a function that allows you to create an instance of an object when it is called. So every time you call it, you get a new object. It looks like that:
```
Comment (simonproffitt) [inline _posts/2018-01-20-constructor-function-and-function-context-in-javascript.md]: ~that~ _this_

**Review by simonproffitt — CHANGES_REQUESTED** [review]
This is really interesting, and something I didn't know anything about. From what you found out, do you think there are any coding best practices that might be adopted to make reading code that uses 'this' less confusing or do you think it's more about really understanding what's going on as you've outlined in this post?

**Comment by caroleolivier** [comment]
It is a good point. ES6 classes really help I think. I added a sentence at the end, lazy me! I could have gone into more details buuuuut I don't feel like it haha.

File: _posts/2018-01-20-constructor-functions-and-function-context-in-javascript.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+First of all, what is a constructor function? A constructor function is a function that allows you to create an instance of an object when it is called. So every time you call it, you get a new object. It looks like this:
+
+```javascript
+function Person(name, age) {
+    this.name = name;
+    this.age = age;
+}
+```
+
+And you call it like with the `new` keyword:
```
Comment (simonproffitt) [inline _posts/2018-01-20-constructor-functions-and-function-context-in-javascript.md]: Sorry, missed this the first time - don't need the 'like' here 'call it with the new keyword'

**Review by simonproffitt — CHANGES_REQUESTED** [review]
One thing I missed then it's good to go - perhaps es6 classes might be a future blog topic?

**Review by simonproffitt — APPROVED** [review]

**Pull request merged**